### PR TITLE
simplify code using optional chaining in multiple config files

### PR DESCRIPTION
**Pull request by Damans227**
### Description

This PR is related to the issue: `ui: Simplify code` #4956

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by Damans227** [comment]
@davidjumani @utchoang Ptal, thx!

**Comment by utchoang** [comment]
@blueorangutan ui

File: ui/src/config/section/infra/clusters.js [hunk 1]
```diff
@@ -114,8 +114,7 @@ export default {
       message: 'label.outofbandmanagement.enable',
       dataView: true,
       show: (record) => {
-        return record.resourcedetails && record.resourcedetails.outOfBandManagementEnabled &&
-          record.resourcedetails.outOfBandManagementEnabled === 'false'
+        return record.resourcedetails?.outOfBandManagementEnabled === 'false'
```
Comment (utchoang) [inline ui/src/config/section/infra/clusters.js:117]: Should be `record?.resourcedetails?.outOfBandManagementEnabled === 'false'`

File: ui/src/config/section/infra/clusters.js [hunk 2]
```diff
@@ -131,8 +130,7 @@ export default {
       message: 'label.outofbandmanagement.disable',
       dataView: true,
       show: (record) => {
-        return !(record.resourcedetails && record.resourcedetails.outOfBandManagementEnabled &&
-          record.resourcedetails.outOfBandManagementEnabled === 'false')
+        return !(record.resourcedetails?.outOfBandManagementEnabled === 'false')
```
Comment (utchoang) [inline ui/src/config/section/infra/clusters.js:133]: Should be `return !(record?.resourcedetails?.outOfBandManagementEnabled === 'false')`

File: ui/src/config/section/infra/clusters.js [hunk 3]
```diff
@@ -148,8 +146,7 @@ export default {
       message: 'label.ha.enable',
       dataView: true,
       show: (record) => {
-        return record.resourcedetails && record.resourcedetails.resourceHAEnabled &&
-          record.resourcedetails.resourceHAEnabled === 'false'
+        return record.resourcedetails?.resourceHAEnabled === 'false'
```
Comment (utchoang) [inline ui/src/config/section/infra/clusters.js:149]: `return record?.resourcedetails?.resourceHAEnabled === 'false'`

File: ui/src/config/section/infra/clusters.js [hunk 4]
```diff
@@ -165,8 +162,7 @@ export default {
       message: 'label.ha.disable',
       dataView: true,
       show: (record) => {
-        return !(record.resourcedetails && record.resourcedetails.resourceHAEnabled &&
-          record.resourcedetails.resourceHAEnabled === 'false')
+        return !(record.resourcedetails?.resourceHAEnabled === 'false')
```
Comment (utchoang) [inline ui/src/config/section/infra/clusters.js:165]: `return !(record?.resourcedetails?.resourceHAEnabled === 'false')`

File: ui/src/config/section/infra/hosts.js [hunk 5]
```diff
@@ -152,8 +152,7 @@ export default {
       docHelp: 'adminguide/hosts.html#out-of-band-management',
       dataView: true,
       show: (record) => {
-        return !(record.outofbandmanagement && record.outofbandmanagement.enabled &&
-          record.outofbandmanagement.enabled === true)
+        return !(record.outofbandmanagement?.enabled === true)
```
Comment (utchoang) [inline ui/src/config/section/infra/hosts.js:155]: `return !(record?.outofbandmanagement?.enabled === true)`

File: ui/src/config/section/infra/hosts.js [hunk 6]
```diff
@@ -188,8 +186,7 @@ export default {
       docHelp: 'adminguide/hosts.html#out-of-band-management',
       dataView: true,
       show: (record) => {
-        return record.outofbandmanagement && record.outofbandmanagement.enabled &&
-          record.outofbandmanagement.enabled === true
+        return record.outofbandmanagement?.enabled === true
```
Comment (utchoang) [inline ui/src/config/section/infra/hosts.js:189]: `return record?.outofbandmanagement?.enabled === true`

File: ui/src/config/section/infra/hosts.js [hunk 7]
```diff
@@ -245,8 +241,7 @@ export default {
       docHelp: 'adminguide/reliability.html#ha-for-hosts',
       dataView: true,
       show: (record) => {
-        return !(record.hostha && record.hostha.haenable &&
-          record.hostha.haenable === true)
+        return !(record.hostha?.haenable === true)
```
Comment (utchoang) [inline ui/src/config/section/infra/hosts.js:244]: `return !(record?.hostha?.haenable === true)`

File: ui/src/config/section/infra/zones.js [hunk 8]
```diff
@@ -126,8 +126,7 @@ export default {
       message: 'label.outofbandmanagement.enable',
       dataView: true,
       show: (record) => {
-        return record.resourcedetails && record.resourcedetails.outOfBandManagementEnabled &&
-          record.resourcedetails.outOfBandManagementEnabled === 'false'
+        return record.resourcedetails?.outOfBandManagementEnabled === 'false'
```
Comment (utchoang) [inline ui/src/config/section/infra/zones.js:129]: `return record?.resourcedetails?.outOfBandManagementEnabled === 'false'`

File: ui/src/config/section/infra/zones.js [hunk 9]
```diff
@@ -143,8 +142,7 @@ export default {
       message: 'label.outofbandmanagement.disable',
       dataView: true,
       show: (record) => {
-        return !(record.resourcedetails && record.resourcedetails.outOfBandManagementEnabled &&
-          record.resourcedetails.outOfBandManagementEnabled === 'false')
+        return !(record.resourcedetails?.outOfBandManagementEnabled === 'false')
```
Comment (utchoang) [inline ui/src/config/section/infra/zones.js:145]: `return !(record?.resourcedetails?.outOfBandManagementEnabled === 'false')`

File: ui/src/config/section/infra/zones.js [hunk 10]
```diff
@@ -160,8 +158,7 @@ export default {
       message: 'label.ha.enable',
       dataView: true,
       show: (record) => {
-        return record.resourcedetails && record.resourcedetails.resourceHAEnabled &&
-          record.resourcedetails.resourceHAEnabled === 'false'
+        return record.resourcedetails?.resourceHAEnabled === 'false'
```
Comment (utchoang) [inline ui/src/config/section/infra/zones.js:161]: `return record?.resourcedetails?.resourceHAEnabled === 'false'`

File: ui/src/config/section/infra/zones.js [hunk 11]
```diff
@@ -177,8 +174,7 @@ export default {
       message: 'label.ha.disable',
       dataView: true,
       show: (record) => {
-        return !(record.resourcedetails && record.resourcedetails.resourceHAEnabled &&
-          record.resourcedetails.resourceHAEnabled === 'false')
+        return !(record.resourcedetails?.resourceHAEnabled === 'false')
```
Comment (utchoang) [inline ui/src/config/section/infra/zones.js:177]: `return !(record?.resourcedetails?.resourceHAEnabled === 'false')`

File: ui/src/config/section/infra/clusters.js (hunk 1 above)
Comment (Damans227) [reply ui/src/config/section/infra/clusters.js:117]: Ok, will apply your suggestion. Thanks for the review.

**Comment by utchoang** [comment]
@blueorangutan ui

**Review by utchoang — APPROVED** [review]
LGTM!

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**